File: arrays-hashing/easy/two-sum/sol.cs
// https://leetcode.com/problems/two-sum/description/

public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> seen = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int c = target - nums[i];
            if (seen.ContainsKey(c))
            {
                return new int[] { seen[c], i };
            }
            seen[nums[i]] = i;
        }
        return null;
    }
}

File: arrays-hashing/easy/two-sum/sol.py
# https://leetcode.com/problems/two-sum/description/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, v in enumerate(nums):
            complement = target - v
            if complement in seen:
                return [seen[complement], i]
            seen[v] = i
